models/errors.go

```diff
@@ -15,15 +15,16 @@ func (err simpleError) Error() string {
 	return err.message
 }
 
+// NewSimpleError returns the custom defined error of type SimpleError.
 func NewSimpleError(msg string) simpleError {
```

Comment:
`type ConversionError struct {
 	simpleError
}`

Creating a new ConversionError, InternalError from other packages do not work because simpleError is unexported..